fabric8/src/main/java/org/jboss/arquillian/ce/fabric8/F8OpenShiftAdapter.java

```diff
@@ -487,14 +493,56 @@ private Object triggerBuild(String namespace, String buildName, String secret, S
     }
 
     protected OpenShiftResourceHandle createResourceFromStream(InputStream stream) throws IOException {
-        ModelNode json;
-        try {
-            json = ModelNode.fromJSONStream(stream);
+    	
+    	try {
+	    	StringBuilder textBuilder = new StringBuilder();
+	        try (Reader reader = new BufferedReader(new InputStreamReader
+	          (stream, Charset.forName(StandardCharsets.UTF_8.name())))) {
+	            int c = 0;
+	            while ((c = reader.read()) != -1) {
+	                textBuilder.append((char) c);
+	            }
+	        }
+	        String content = textBuilder.toString();
+	       
+	        char nonwhitespace;
+	        int index=0;
+	        do {
+	        	nonwhitespace=content.charAt(index++);
+	        } while (index < content.length() && Character.isWhitespace(nonwhitespace));
+	        
+	        if ( nonwhitespace == '{' ) {
+	        	ModelNode json;
```

Comment:
This logic is not complete. "5", or "[]" are valid JSON strings. Why not use the parser to figure out whether the content is JSON or not? It would save us some lines of code as well. See http://stackoverflow.com/questions/3679479/check-if-file-is-json-java